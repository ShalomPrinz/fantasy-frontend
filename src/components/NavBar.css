.filter-invert {
    filter: invert(1);
}

.nav-links:hover {
    color: #ffdd40 !important;
}

.navbar-brand {
  color: white !important;
}

.navbar {
    background-color: #0E1E5B;
    color: white;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1.2rem;
    position: sticky;
    top: 0;
    vertical-align: middle;
  }
  
  .nav-links {
    color: #fff;
    text-decoration: none;
    padding: 0.5rem 1rem;
    height: 100%;
    border-bottom: 3px solid transparent;
  }
  
  .nav-item {
    line-height: 40px;
    margin-right: 1rem;
  }
  
  .nav-item:after {
    content: "";
    display: block;
    height: 3px;
    width: 0;
    background: transparent;
    transition: width 0.7s ease, background-color 0.5s ease;
  }
  
  .nav-item:hover:after {
    width: 100%;
    background: #ffdd40;
  }
  
  .nav-item .active {
    color: #ffdd40;
    border: 1px solid #ffdd40;
  }
  
  @media screen and (max-width: 960px) {
    .nav-item .active {
      color: #ffdd40;
      border: none;
    }
    .nav-links {
      padding: 1.5rem;
      width: 100%;
      display: table;
    }
  }