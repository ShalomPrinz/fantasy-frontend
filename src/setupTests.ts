// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import axios from "axios";
import { setupServer } from "msw/node";

import { handlers } from "../test";
import * as TeamContext from "./contexts/TeamContext";
import * as UserContext from "./features/authentication/contexts/UserContext";
import { LeagueInfo, Message, Player, Team, User } from "./types";

jest.setTimeout(10000);

axios.defaults.baseURL = "";

const server = setupServer(...handlers);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

interface MockedUserContext {
  logout?: () => Promise<void>;
  state?: UserContext.UserState;
  user?: User;
}

interface MockedUser {
  inbox?: Message[];
  leagues?: LeagueInfo[];
  name?: string;
  players?: Player[];
}

export const getTestUser = (user?: MockedUser): User => ({
  id: "id",
  inbox: user?.inbox || [],
  leagues: user?.leagues || [],
  name: user?.name || "",
  team: new Team(user?.players || []),
});

export const mockUser = (context: MockedUserContext) => {
  const userContext = jest.spyOn(UserContext, "useUser");

  const { logout, state, user } = context;
  const mockedLogout = typeof logout !== "undefined" ? logout : async () => {};
  const mockedState =
    typeof state !== "undefined" ? state : UserContext.UserState.LOGGED_USER;
  const mockedUser = typeof user !== "undefined" ? user : getTestUser();

  userContext.mockImplementation(() => ({
    loading: false,
    login: async () => {},
    logout: mockedLogout,
    register: async () => {},
    state: mockedState,
    user: mockedUser,
    invalidate: async () => {},
  }));

  return userContext.mockRestore;
};

export const mockTeam = (team: Team) => {
  const teamContext = jest.spyOn(TeamContext, "useTeamState");
  teamContext.mockImplementation(() => team);
};

export * from "../test";
