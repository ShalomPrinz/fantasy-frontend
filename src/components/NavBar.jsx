import React, { useState } from "react";
import { NavLink } from "react-router-dom";
import Image from "react-bootstrap/Image";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";

import logo from "../res/cl_logo.png";

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes, faBars } from '@fortawesome/free-solid-svg-icons';
import "./NavBar.css";

const pages = [
  {
    name: "My Team",
    url: "/team"
  },
  {
    name: "Second",
    url: "/"
  },
  {
    name: "Third",
    url: "/"
  }
]

let key = 0;

function NavBar() {
  const [expanded, setExpanded] = useState(false);

  return (
    <Navbar className="navbar bg-default" expand="lg" expanded={expanded}>
      <NavLink className="navbar-brand fs-1" to="/">
        <Image 
          height="90px"
          className="p-2 m-3 filter-invert" 
          src={logo} 
          title="Champions League" 
        />
        Fantasy
      </NavLink>
      <Navbar.Toggle
        aria-controls="responsive-navbar-nav"
        onClick={() => setExpanded(!expanded)}
      >
        {expanded && <FontAwesomeIcon color="white" icon={faTimes} />}
        {!expanded && <FontAwesomeIcon color="white" icon={faBars} />}
      </Navbar.Toggle>
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav
          onClick={() => {
            if (expanded) setExpanded(!expanded);
          }}
        >
          {pages.map((page) => (
            <NavLink className="nav-item nav-links fw-light" to={page.url} key={key++}>
              {page.name}
            </NavLink>
          ))}
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
}

export default NavBar;
