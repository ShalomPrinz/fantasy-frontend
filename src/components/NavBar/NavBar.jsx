import React, { useCallback, useState } from "react";

import { faBars, faTimes } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Image from "react-bootstrap/Image";
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import { NavLink } from "react-router-dom";

import { cl_logo } from "../../res";
import ConditionalList from "../ConditionalList";
import "./NavBar.css";

const pages = [
  {
    name: "My Team",
    url: "/team"
  },
  {
    name: "Second",
    url: "/"
  },
  {
    name: "Third",
    url: "/"
  }
]

function NavBar() {
  const [expanded, setExpanded] = useState(false);

  const pageCallback = useCallback((page) => (
    <NavLink className="nav-item nav-links fw-light" to={page.url}>
      {page.name}
    </NavLink>
  ), []);

  const icon = expanded ? faTimes : faBars;

  return (
    <Navbar className="navbar bg-default" expand="lg" expanded={expanded}>
      <NavLink className="navbar-brand fs-1" to="/">
        <Image 
          height="90px"
          className="p-2 m-3 filter-invert" 
          src={cl_logo} 
          title="Champions League" 
        />
        Fantasy
      </NavLink>
      <Navbar.Toggle
        aria-controls="responsive-navbar-nav"
        onClick={() => setExpanded(!expanded)}
      >
        <FontAwesomeIcon color="white" icon={icon} />
      </Navbar.Toggle>
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav onClick={() => expanded && setExpanded(!expanded)}>
          <ConditionalList itemCallback={pageCallback} list={pages} />
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
}

export default NavBar;
