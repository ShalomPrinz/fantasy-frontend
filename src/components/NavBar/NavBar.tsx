import { useCallback, useState } from "react";

import { faBars, faTimes } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import Image from "react-bootstrap/Image";
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import { NavLink } from "react-router-dom";

import { cl_logo } from "../../res";
import ConditionalList from "../ConditionalList";
import "./NavBar.css";

interface Page {
  id: number;
  name: string;
  url: string;
}

const pages = [
  {
    id: 0,
    name: "My Team",
    url: "/team",
  },
  {
    id: 1,
    name: "Second",
    url: "/",
  },
  {
    id: 2,
    name: "Third",
    url: "/",
  },
];

function NavBar() {
  const [expanded, setExpanded] = useState(false);

  const pageCallback = useCallback(
    ({ name, url }: Page) => (
      <NavLink className="nav-item nav-links fw-light" to={url}>
        {name}
      </NavLink>
    ),
    []
  );

  const icon = expanded ? faTimes : faBars;
  const setNotExpanded = () => expanded && setExpanded(!expanded);

  return (
    <Navbar className="navbar bg-default" expand="lg" expanded={expanded}>
      <NavLink onClick={setNotExpanded} className="navbar-brand fs-1" to="/">
        <Image
          height="90px"
          className="p-2 m-3 filter-invert"
          src={cl_logo}
          title="Champions League"
        />
        Fantasy
      </NavLink>
      <Navbar.Toggle
        aria-controls="responsive-navbar-nav"
        onClick={() => setExpanded(!expanded)}
      >
        <FontAwesomeIcon color="white" icon={icon} />
      </Navbar.Toggle>
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav onClick={setNotExpanded}>
          <ConditionalList itemCallback={pageCallback} list={pages} />
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
}

export default NavBar;
