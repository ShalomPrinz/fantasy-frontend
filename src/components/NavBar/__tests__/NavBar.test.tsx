import { BrowserRouter } from "react-router-dom";
import { clickElement, render, screen } from "setupTests";

import NavBar from "../NavBar";

const Component = (
  <BrowserRouter>
    <NavBar />
  </BrowserRouter>
);

describe("NavBar", () => {
  it("should render NavBar component", () => {
    const { asFragment } = render(Component);

    expect(asFragment()).toMatchSnapshot();
  });

  describe("when navbar expands", () => {
    it("should change navbar layout", async () => {
      const { user, asFragment } = render(Component);

      expect(asFragment()).toMatchSnapshot();

      const element = screen.queryByRole("button");
      if (element) await clickElement(user, element);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should allow collapse via main link", async () => {
      const { user, asFragment } = render(Component);

      const firstRender = asFragment();
      expect(firstRender).toMatchSnapshot();

      const mainLink = screen.queryAllByRole("link")[0];
      if (mainLink) await clickElement(user, mainLink);

      expect(asFragment()).toEqual(firstRender);

      const toggleButton = screen.queryByRole("button");
      if (toggleButton) await clickElement(user, toggleButton);

      expect(asFragment()).not.toEqual(firstRender);

      if (mainLink) await clickElement(user, mainLink);

      // This differs a little from firstRender
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
