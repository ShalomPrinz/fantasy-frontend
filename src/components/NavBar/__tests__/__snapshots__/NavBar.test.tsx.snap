// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render NavBar component 1`] = `
<DocumentFragment>
  <nav
    class="navbar bg-default navbar navbar-expand-lg navbar-light"
  >
    <a
      aria-current="page"
      class="navbar-brand fs-1 active"
      href="/"
    >
      <img
        class="p-2 m-3 filter-invert"
        height="90px"
        src="cl_logo.png"
        title="Champions League"
      />
      Fantasy
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars "
        color="white"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-links fw-light"
          href="/team"
        >
          My Team
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Second
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Third
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavBar when navbar expands should allow collapse via main link 1`] = `
<DocumentFragment>
  <nav
    class="navbar bg-default navbar navbar-expand-lg navbar-light"
  >
    <a
      aria-current="page"
      class="navbar-brand fs-1 active"
      href="/"
    >
      <img
        class="p-2 m-3 filter-invert"
        height="90px"
        src="cl_logo.png"
        title="Champions League"
      />
      Fantasy
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars "
        color="white"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-links fw-light"
          href="/team"
        >
          My Team
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Second
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Third
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavBar when navbar expands should allow collapse via main link 2`] = `
<DocumentFragment>
  <nav
    class="navbar bg-default navbar navbar-expand-lg navbar-light"
  >
    <a
      aria-current="page"
      class="navbar-brand fs-1 active"
      href="/"
    >
      <img
        class="p-2 m-3 filter-invert"
        height="90px"
        src="cl_logo.png"
        title="Champions League"
      />
      Fantasy
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars "
        color="white"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="navbar-collapse collapsing"
      id="responsive-navbar-nav"
      style=""
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-links fw-light"
          href="/team"
        >
          My Team
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Second
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Third
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavBar when navbar expands should change navbar layout 1`] = `
<DocumentFragment>
  <nav
    class="navbar bg-default navbar navbar-expand-lg navbar-light"
  >
    <a
      aria-current="page"
      class="navbar-brand fs-1 active"
      href="/"
    >
      <img
        class="p-2 m-3 filter-invert"
        height="90px"
        src="cl_logo.png"
        title="Champions League"
      />
      Fantasy
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars "
        color="white"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-links fw-light"
          href="/team"
        >
          My Team
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Second
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Third
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavBar when navbar expands should change navbar layout 2`] = `
<DocumentFragment>
  <nav
    class="navbar bg-default navbar navbar-expand-lg navbar-light"
  >
    <a
      aria-current="page"
      class="navbar-brand fs-1 active"
      href="/"
    >
      <img
        class="p-2 m-3 filter-invert"
        height="90px"
        src="cl_logo.png"
        title="Champions League"
      />
      Fantasy
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-xmark "
        color="white"
        data-icon="xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="navbar-collapse collapsing"
      id="responsive-navbar-nav"
      style="height: 0px;"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-links fw-light"
          href="/team"
        >
          My Team
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Second
        </a>
        <a
          aria-current="page"
          class="nav-item nav-links fw-light active"
          href="/"
        >
          Third
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
